doc/developer_guide.md

```diff
@@ -0,0 +1,237 @@
+# Setting up a development environment
```

Comment:
Have you checked if this looks good on the webpage?

Reply:
LGTM!

![image](https://github.com/user-attachments/assets/891279f8-12bc-4c6d-8fe0-570412e5cba9)